File: server/game.go
package tcabaagno

type Game struct {
	Forms []QAForm
}

type Questions struct {
	Who   string
	What  string
	When  string
	Where string
	Why   string
}

type Answers struct {
	Who   string
	What  string
	When  string
	Where string
	Why   string
}

type QAForm struct {
	Questions Questions
	Answers   Answers
	Player    string
}

File: ui/src/app/models/game.ts
export class Game {
    id: string;
    status: string;
}

export class RegistrationResponse {
    id: string;
    gameId: string;
}

export class GameInfo {
    id: string;
    numPlayers: number;
}

export class SubmitQuestionsPayload {
    who: string;
    what: string;
    when: string;
    where: string;
    why: string;
}

export class Message<T> {
    type: string;
    payload: T;

    constructor(type: string, payload?: T) {
        this.type = type;
        if (payload) {
            this.payload = payload;
        }
    }
}
